packages/shared/lib/core/wallet/utils/getTokenMetadataFromFoundryOutput.ts

```diff
@@ -1,24 +1,70 @@
-import { ITokenMetadata } from '../interfaces'
+import { IIrc30Metadata, IPersistedAsset } from '../interfaces'
 import { selectedAccount } from '@core/account'
 import { get } from 'svelte/store'
 import { IFoundryOutput } from '@iota/types'
 import { Converter } from '@lib/converter'
+import { addPersistedAsset, getPersistedAsset } from '../stores/persisted-assets.store'
 
-export async function getTokenMetadataFromFoundryOutput(tokenId: string): Promise<ITokenMetadata> {
+export function getAssetFromPersistedAssets(tokenId: string): IPersistedAsset {
```

Comment:
Lets extract all these functions into there own files, and put them in the correct actions, utils or helpers folder